src/components/overview/chart/index.tsx

```diff
@@ -0,0 +1,119 @@
+import { colors } from '../../../theme';
+import { data } from './helper';
+import {
+  Area,
+  AreaChart,
+  CartesianGrid,
+  ResponsiveContainer,
+  Tooltip,
+  XAxis,
+  YAxis,
+} from 'recharts';
+import { FunctionComponent, useState } from 'react';
+const Chart = () => {
+  const currentDate = new Date();
+  const currentHour = currentDate.getHours();
+  const [isTodayHovered, setIsTodayHovered] = useState(false);
+  const [isYesterdayHovered, setIsYesterdayHovered] = useState(false);
+  interface IProps {
+    cx: number;
+    cy: number;
+    gray?: boolean;
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const CustomizedDot: FunctionComponent<IProps & any> = ({ cx, cy, gray }) => {
+    return (
+      <svg x={cx - 13} y={cy - 13} height='100' width='100'>
+        <circle
+          cx='13'
+          cy='13'
+          r='5'
+          stroke={gray ? `${colors.grayDark}` : `${colors.primaryBlue}`}
+          strokeWidth='4'
+          fill='white'
+        />
+        <circle
+          cx='13'
+          cy='13'
+          r='12'
+          stroke={gray ? `${colors.grayDark}` : `${colors.primaryBlue}`}
+          opacity='0.16'
+          strokeWidth='2'
+          fill='transparent'
+        />
+      </svg>
+    );
+  };
+  const filteredData = data.map((entry) => ({
+    ...entry,
+    Today: entry.Today && parseInt(entry.hours) < currentHour ? entry.Today : null,
```

Comment:
What this filter does?

Reply:
It filters the data only before the current hour - so that the graph shows the data on the current hour(like on Figma) e.g. if now is 18:44 it will show info only for 1-18 hours)